synfig-studio/src/gui/dialogs/vectorizersettings.h

```diff
@@ -104,7 +103,10 @@ class VectorizerSettings : public Gtk::Dialog
     //                    : (VectorizerConfiguration *)new CenterlineConfiguration(getCenterlineConfiguration());
   	// }
 	
-private:
+private:	
+	void initialize_parameters(etl::handle<synfig::Layer_Bitmap> my_layer_bitmap,
+		etl::handle<studio::Instance> selected_instance,std::unordered_map <std::string,int>& configmap, etl::handle<synfig::Layer> reference_layer);
```

Comment:
You first passed those arguments from app.cpp to `VectorizerSettings::create(...)` and then you passed them again to `initialize_parameters(...)` so isn't it safe to just specify all arguments for 2nd function as references instead of passing them by values? Since we know it's safe to modify their values without affecting the original data passed from 1st function and... save memory(???) I don't know how big the data is so for forgive my dumb assumption.

Reply:
Oh right, it should be references. I just copied the function headers and slightly modified them, so yes this would be a waste of memory. Thanks for the tip!